pytorch_lightning/utilities/parsing.py

```diff
@@ -174,75 +174,61 @@ def __repr__(self):
         return out
 
 
-def lightning_hasattr(model, attribute):
-    """ Special hasattr for lightning. Checks for attribute in model namespace,
-        the old hparams namespace/dict, and the datamodule. """
+def lightning_get_attr_holder(model, attribute):
+    """ Gets the object or dict that holds attribute, or None. Checks for attribute in model namespace,
+            the old hparams namespace/dict, and the datamodule, returns the first one that has it. """
     trainer = model.trainer
 
     # Check if attribute in model
     if hasattr(model, attribute):
-        attr = True
+        return model
+
     # Check if attribute in model.hparams, either namespace or dict
-    elif hasattr(model, 'hparams'):
-        if isinstance(model.hparams, dict):
-            attr = attribute in model.hparams
-        else:
-            attr = hasattr(model.hparams, attribute)
+    if hasattr(model, 'hparams'):
+        if isinstance(model.hparams, dict) and attribute in model.hparams:
+            return model.hparams
+        elif hasattr(model.hparams, attribute):
+            return model.hparams
+
     # Check if the attribute in datamodule (datamodule gets registered in Trainer)
-    elif trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
-        attr = getattr(trainer.datamodule, attribute)
-    else:
-        attr = False
+    if trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
+        if getattr(trainer.datamodule, attribute):
+            return trainer.datamodule
```

Comment:
Isn't this already covered by `and hasattr(trainer.datamodule, attribute)`?

```suggestion
        return trainer.datamodule
```